src/stores/messages/MessagesDataProviderStore.ts

```diff
@@ -181,12 +198,24 @@ export default class MessagesDataProviderStore {
 	};
 
 	@action
-	public startPreviousMessagesChannel = (query: MessagesSSEParams) => {
+	public startPreviousMessagesChannel = (query: MessagesSSEParams, interval?: number) => {
+		this.noMatchingMessagesPrev = false;
+		this.prevLoadEndTimestamp = null;
+
 		this.searchChannelPrev = new SSEChannel(
 			this.messagesStore.filterStore.messsagesSSEConfig.type,
 			query,
 			this.onPrevChannelResponse,
 			this.onLoadingError,
+			interval
+				? e => {
+						if (query.startTimestamp - e.timestamp > interval * 1000 * 60) {
```

Comment:
actually we cant compare `query.startTimestamp` with keep_alive timestamp.
Because startTimestamp is initialized once when we start loading, then when we scroll and load more we change only `resumeFromId`.